### PR TITLE
[Doc] Add a tutorial for nuScenes dataset

### PR DESCRIPTION
**Pull request by Tai-Wang**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Please describe the motivation of this PR and the goal you want to achieve through this PR.
Add a tutorial for nuScenes dataset.

## Modification
Please briefly describe what modification is made in this PR.
1. Add a tutorial for nuScenes dataset.
2. Add a Chinese placeholder for the nuScenes tutorial.

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.
No.

## Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/datasets/nuscenes_det.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        We maintain a default attribute collection and mapping for attribute classification.
+        Please refer to [here](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/datasets/nuscenes_mono_dataset.py#L53) for more details.
+        - info['annotations'][i]['id']: Annotation id. Defaults to `i`.
+
+Here we only explain the data recorded in the training info files. The same applies to validation and testing set.
+
+The core function to get `nuscenes_infos_xxx.pkl` and `nuscenes_infos_xxx_mono3d.coco.json` are [\_fill_trainval_infos](https://github.com/open-mmlab/mmdetection3d/blob/master/tools/data_converter/nuscenes_converter.py#L143) and [get_2d_boxes](https://github.com/open-mmlab/mmdetection3d/blob/master/tools/data_converter/nuscenes_converter.py#L397), respectively.
+Please refer to [nuscenes_converter.py](https://github.com/open-mmlab/mmdetection3d/blob/master/tools/data_converter/nuscenes_converter.py) for more details.
+
+## Train pipeline
```
Comment (Wuziyi616) [inline docs/datasets/nuscenes_det.md:130]: `Train` --> `Training`

File: docs/datasets/nuscenes_det.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+### LiDAR-Based Methods
+
+A typical train pipeline of LiDAR-based 3D detection (including multi-modality methods) on nuScenes is as below.
```
Comment (Wuziyi616) [inline docs/datasets/nuscenes_det.md:134]: Also `training` here would be better?

File: docs/datasets/nuscenes_det.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+```
+
+Compared to general cases, nuScenes has a specific `'LoadPointsFromMultiSweeps'` pipeline to load point clouds from consecutive frames. This is a common practice used in this setting.
+Please refer to the nuScenes [original paper](https://arxiv.org/abs/1903.11027) for more details.
+The default `use_dim` in `'LoadPointsFromMultiSweeps'` is `[0, 1, 2, 4]`, where the first 3 dimensions refer to point coordinates and the last refers to timestamp differences.
+Intensity is not used by default due to its yielded noise when concatenating the points from different frames.
+
+### Vision-Based Methods
+
+A typical train pipeline of image-based 3D detection on nuScenes is as below.
```
Comment (Wuziyi616) [inline docs/datasets/nuscenes_det.md:171]: `training`

File: docs/datasets/nuscenes_det.md [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+- It needs to load 3D annotations.
+- Some data augmentation techniques need to be adjusted, such as `RandomFlip3D`.
+Currently we do not support more augmentation methods, because how to transfer and apply other techniques is still under explored.
+
+## Evaluation
+
+An example to evaluate PointPillars with 8 GPUs with nuScenes metrics is as follows
+
+```shell
+bash configs/pointpillars/hv_pointpillars_fpn_sbn-all_4x8_2x_nus-3d.py checkpoints/hv_pointpillars_fpn_sbn-all_4x8_2x_nus-3d_20200620_230405-2fa62f3d.pth 8 --eval bbox
```
Comment (Wuziyi616) [inline docs/datasets/nuscenes_det.md:209]: The eval code missed something? Shouldn't it be `bash tools/dist_test.py configs/xxxxx`

File: docs/datasets/nuscenes_det.md [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+- `nuscenes_infos_train.pkl`: training dataset infos, each frame info has two keys: `metadata` and `infos`.
+`metadata` contains the basic information for the dataset itself, such as `{'version': 'v1.0-trainval'}`, while `infos` contains the detailed information as follows:
+    - info['lidar_path']: The file path of the lidar point cloud data.
+    - info['token']: Sample data token.
+    - info['sweeps']: Sweeps information (`sweeps` in the nuScenes refer to the intermediate frames without annotations, while `samples` refer to those key frames with annotations).
+        - info['sweeps'][i]['data_path']: The data path of i-th sweep.
+        - info['sweeps'][i]['type']: The sweep data type, e.g., `'lidar'`.
+        - info['sweeps'][i]['sample_data_token']: The sweep sample data token.
+          - info['sweeps'][i]['sensor2ego_translation']: The translation from the current sensor (for collecting the sweep data) to ego vehicle. (1x3 list)
+          - info['sweeps'][i]['sensor2ego_rotation']: The rotation from the current sensor (for collecting the sweep data) to ego vehicle. (1x4 list in the quaternion format)
```
Comment (Wuziyi616) [inline docs/datasets/nuscenes_det.md:75]: The indent of line 74-75 is a bit strange?

**Review by Wuziyi616 — COMMENTED** [review]
Do you think it's reasonable to explain a bit about nuScenes' coordinate system, e.g. `NuscBox`'s definition with our coordinate system? There are several related operation in the codebase. I understand that after refactoring the hack of rotation, the related code is less now. They exist in, e.g. `nuscenes_converter.py` and `output_to_nusc_box` in nusc_mono_dataset.

**Comment by Tai-Wang** [comment]
> Do you think it's reasonable to explain a bit about nuScenes' coordinate system, e.g. `NuscBox`'s definition with our coordinate system? There are several related operation in the codebase. I understand that after refactoring the hack of rotation, the related code is less now. They exist in, e.g. `nuscenes_converter.py` and `output_to_nusc_box` in nusc_mono_dataset.

Yes, I add a "Notes" section to explain questions like that.

**Review by Wuziyi616 — APPROVED** [review]
Great tutorial!

**Pull request merged**